### ISSUE TITLE
no idea to support java 9+ feature "Multi-Release"

### ISSUE THREAD

**Issue by ImFl0wow** [radioegor146/native-obfuscator#71]
MANIFEST.MF
Multi-Release: true

multirelease.jar
├── META-INF
│   └── versions
│       └── 9
│           └── multirelease
│               └── Helper.class
├── multirelease
    ├── Helper.class
    └── Main.class

**Comment by radioegor146** [comment]
Wouldn't it be easier to build two separate jars and then combine them as needed?

**Comment by ImFl0wow** [comment]
agree&nbsp;


曼珠沙华
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Egor ***@***.***&gt;; 
发送时间: 2024年1月16日(星期二) 下午2:16
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [radioegor146/native-obfuscator] no idea to support java 9+ feature &quot;Multi-Release&quot; (Issue #71)





 
Wouldn't it be easier to build two separate jars and then combine them as needed?

**Comment by ImFl0wow** [comment]
You are right, but that feature this is usually used for utilities class, if a different VM version has a more efficient method or the method is deprecated, then the VM will automatically choose the class in the versions folder. if you do it like you said, it will make referencing the dependency complicated.
